packages/image/src/Image.svelte

```diff
@@ -0,0 +1,173 @@
+		if (!fixed) {
+			return { widths: device_sizes.filter((s) => s <= width * 2), kind: 'w' };
+		}
+
+		const widths = [
+			...new Set(
+				// > This means that most OLED screens that say they are 3x resolution,
+				// > are actually 3x in the green color, but only 1.5x in the red and
+				// > blue colors. Showing a 3x resolution image in the app vs a 2x
+				// > resolution image will be visually the same, though the 3x image
+				// > takes significantly more data. Even true 3x resolution screens are
+				// > wasteful as the human eye cannot see that level of detail without
+				// > something like a magnifying glass.
+				// https://blog.twitter.com/engineering/en_us/topics/infrastructure/2019/capping-image-fidelity-on-ultra-high-resolution-devices.html
+				[width, width * 2].map(
+					(w) => all_sizes.find((p) => p >= w) || all_sizes[all_sizes.length - 1]
+				)
+			)
+		];
+		return { widths, kind: 'x' };
+	}
+
+	/**
+	 * @param {string} src
+	 */
+	function matches_domain(src) {
+		const url = new URL(src, 'http://n'); // if src is protocol relative, use dummy domain
+		if (url.href === src) {
+			return domains.some((domain) => url.hostname === domain);
+		} else {
+			return true; // relative urls are always ok
+		}
+	}
+</script>
+
+<script>
+	/** @type {string | import('vite-imagetools').Img | import('vite-imagetools').Picture} */
+	export let src;
+	/** @type {string} */
+	export let alt;
+	/** @type {number}*/
+	export let width;
+	/** @type {number}*/
+	export let height;
+	/** @type {string | undefined} */
+	export let sizes = undefined;
+	/** @type {string | undefined} */
+	export let style = undefined;
+	/** @type {string} */
+	export let provider = 'default';
+	/** @type {any} */
+	export let providerOptions = undefined;
+	/** @type {'lazy' | 'eager' | undefined} */
+	export let loading = undefined;
+	/** @type {boolean} */
+	export let priority = false;
+
+	let srcset = '';
+	let _src = '';
+
+	$: {
+		if (typeof src !== 'string') {
+			srcset = (src.img || src).srcset?.map((i) => `${i.src} ${i.w}w`)?.join(', ');
+			_src = (src.img || src).src;
+			width = width || (src.img || src).w;
+			height = height || (src.img || src).h;
+		} else {
+			if (matches_domain(src)) {
+				const p = providers[provider];
+				if (DEV && !p) {
+					throw new Error(
+						`No provider named "${provider}" found. Available providers are ${Object.keys(
+							providers
+						).join(', ')}. Check the kit.images.providers configuration in your svelte.config.js.`
+					);
+				}
+
+				const widths = getWidths(width, !!style && /width: \d+(px|em|rem)/.test(style), sizes);
+				srcset = widths.widths
+					.map((w) => {
+						const url = DEV
+							? src
+							: p.getURL({
+									src: /** @type {string} */ (src),
+									width: w,
+									height: Math.round(w * (width / height)),
+									options: providerOptions
+							  });
+						return `${url} ${w}${widths.kind}`;
+					})
+					.join(', ');
+				_src = srcset.at(-1) ?? src;
```

Comment:
does this work? if srcset.at(-1) contains a comma separated list with dimensions, it's not a plain src.

If the set is ordered by size, you want the first or last one, depending on quality. I'd pick highest quality
```suggestion
				_src = srcset?.at(-1).split(',').at(-1).split(' ',1).at(0) ?? src;
```